### PR TITLE
Task 4: Map (new class) #4; Task 5: Map (constructor) #5; Task 6: Map (getIsland) #9

### LINKED ISSUE #4: Task 4: Map (new class)
Napisz klasę Map, która będzie posiadała std::vector<Island&> przechowujący wszystkie wyspy na mapie, oraz zmienną Island* currentPosition_ określającą aktualną pozycję gracza na mapie.

### LINKED ISSUE #5: Task 5: Map (constructor)
W klasie Map utwórz konstruktor bezargumentowy, a w jego ciele utwórz 10 wysp, które przechowasz w std::vector<Island&>. Do wygenerowania losowych wartości pozycji wysp na mapie skorzystaj z przykładu na cppreference. Wymyśl sposób, aby pozycje wysp się nie powielały.

### LINKED ISSUE #9: Task 6: Map (getIsland)
W klasie Map napisz funkcję

Island* getIsland(const Island::Coordinates& coordinate)

Powinna ona przeszukać std::vector<Island&> i zwrócić szukaną wyspę.

### PR DESCRIPTION
**Pull request by madebykj**

### CODE REVIEW

File: shm/Map.cpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    std::uniform_int_distribution<> distrib(randomNumberStart, maxPositionXY);
+    vecOfIslands_.reserve(numberOfIslands);
+    int notRepeatedPositions = 0;
+    while (notRepeatedPositions != 10) {
+        int positionX = distrib(gen);
+        int positionY = distrib(gen);
+        if (std::none_of(vecOfIslands_.begin(), vecOfIslands_.end(), [positionX, positionY](auto& i) {
+                return i.getPosition() == Coordinates(positionX, positionY);
+            })) {
+            vecOfIslands_.emplace_back(Island(positionX, positionY));
```
Comment (lksobczyk) [inline shm/Map.cpp]: Excellently

**Review by lksobczyk — CHANGES_REQUESTED** [review]
Could you use prefix instead of suffix for fields in class?

**Review by lksobczyk — COMMENTED** [review]
Also add "Map.o" to Makefile

**Review by lksobczyk — APPROVED** [review]
Well done.

File: shm/Map.cpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    while (notRepeatedPositions != 10) {
+        int positionX = distrib(gen);
+        int positionY = distrib(gen);
+        if (std::none_of(_vecOfIslands.begin(), _vecOfIslands.end(), [positionX, positionY](auto& i) {
+                return i.getPosition() == Coordinates(positionX, positionY);
+            })) {
+            _vecOfIslands.emplace_back(Island(positionX, positionY));
+            notRepeatedPositions++;
+        }
+    }
```
Comment (kszytko) [inline shm/Map.cpp]: It's a good place for future refactoring.
You have constant value in header file `numberOfIslands` - use it instead `!= 10`

File: shm/Map.hpp [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <vector>
+
+#include "Island.hpp"
+
+constexpr int randomNumberStart = 1;
```
Comment (kszytko) [inline shm/Map.hpp]: It's better to have one naming concept. 
```suggestion
constexpr int minPositionXY = 1;
```

**Review by kszytko — CHANGES_REQUESTED** [review]

**Review by lksobczyk — APPROVED** [review]

**Review by kszytko — APPROVED** [review]
Good

**Pull request merged**